site/src/api/queries/organizations.ts

```diff
@@ -156,28 +158,91 @@ export const organizationPermissions = (organizationId: string | undefined) => {
-              organization_id: organizationId,
-            },
-            action: "read",
+        },
+      }),
+  };
+};
+
+/**
+ * Fetch permissions for all provided organizations.
+ *
+ * If organizations are undefined, return a disabled query.
+ */
+export const organizationsPermissions = (
+  organizations: Organization[] | undefined,
+) => {
+  if (!organizations) {
+    return { enabled: false };
+  }
+
+  return {
+    queryKey: ["organizations", "permissions"],
+    queryFn: async () => {
+      // Only request what we need for the sidebar, which is one edit permission
+      // per sub-link (audit page, settings page, groups page, and members page)
+      // that tells us whether to show that page, since we only show them if you
+      // can edit (and not, at the moment if you can only view).
+      const checks = (organizationId: string) => ({
+        editMembers: {
+          object: {
+            resource_type: "organization_member",
+            organization_id: organizationId,
           },
-          editOrganization: {
-            object: {
-              resource_type: "organization",
-              organization_id: organizationId,
-            },
-            action: "update",
+          action: "update",
+        },
+        editGroups: {
+          object: {
+            resource_type: "group",
+            organization_id: organizationId,
           },
-          auditOrganization: {
-            object: {
-              resource_type: "audit_log",
-              organization_id: organizationId,
-            },
-            action: "read",
+          action: "update",
+        },
+        editOrganization: {
+          object: {
+            resource_type: "organization",
+            organization_id: organizationId,
           },
+          action: "update",
         },
-      }),
+        auditOrganization: {
+          object: {
+            resource_type: "audit_log",
+            organization_id: organizationId,
+          },
+          action: "read",
+        },
+      });
+
+      // The endpoint takes a flat array, so to avoid collisions prepend each
+      // check with the org ID (the key can be anything we want).
+      const prefixedChecks = organizations
+        .map((org) =>
+          Object.entries(checks(org.id)).map(([key, val]) => [
+            `${org.id}.${key}`,
+            val,
+          ]),
+        )
+        .flat();
+
+      const response = await API.checkAuthorization({
+        checks: Object.fromEntries(prefixedChecks),
+      });
+
+      // Now we can unflatten by parsing out the org ID from each check.
+      return Object.entries(response).reduce(
+        (acc, [key, value]) => {
+          const index = key.indexOf(".");
+          const orgId = key.substring(0, index);
+          const perm = key.substring(index + 1);
+          if (!acc[orgId]) {
+            acc[orgId] = { [perm]: value };
+          } else {
+            acc[orgId][perm] = value;
+          }
```

Comment:
```suggestion
          if (!acc[orgId]) {
            acc[orgId] = {};
          }
          acc[orgId][perm] = value;
```

Reply:
Truuuuueeeee this is probably better lol, I had it like that and changed it but for the life of me I cannot justify it.

I also considered:

```
acc[orgId] ??= {}
acc[orgId][perm] = value;
```

But I think maybe it looks too magical